tests/Filter/StringFilterTest.php

```diff
@@ -34,50 +48,133 @@ public function testEmpty(): void
+            0,
+            false,
+            ['case_sensitive' => true],
+            ['binary' => Binary::class],
+        ];
+        yield 'filter by 0' => [['WHERE alias.field LIKE :field_name_0'], 0, true];
+        yield 'not filter by \'0\'' => [['WHERE alias.field LIKE :field_name_0'], '0', false];
+        yield 'filter by \'0\'' => [['WHERE alias.field LIKE :field_name_0'], '0', true];
     }
 
     /**
-     * @dataProvider getValues
+     * @dataProvider getValuesForDefaultType
      */
-    public function testDefaultType($value, bool $allowEmpty): void
+    public function testDefaultType(array $expected, $value, bool $allowEmpty, array $options = [], array $dqlExtensions = []): void
     {
-        $filter = new StringFilter();
-        $filter->initialize('field_name', ['allow_empty' => $allowEmpty]);
+        $doctrineConfig = $this->createMock(Configuration::class);
+        $doctrineConfig->expects($this->exactly(!($options['case_sensitive'] ?? null) || '' === (string) $value ? 0 : 1))
+            ->method('getCustomStringFunction')
+            ->with('binary')
+            ->willReturn($dqlExtensions['binary'] ?? null);
+
+        $em = $this->createMock(EntityManagerInterface::class);
+        $em->expects($this->once())
+            ->method('getConfiguration')
+            ->willReturn($doctrineConfig);
+
+        $filter = new StringFilter($em);
+        $filter->initialize('field_name', ['allow_empty' => $allowEmpty] + $options);
 
         $proxyQuery = new ProxyQuery($this->createQueryBuilderStub());
         $this->assertSameQuery([], $proxyQuery);
 
         $filter->filter($proxyQuery, 'alias', 'field', ['value' => $value, 'type' => null]);
 
+        $this->assertSameQuery($expected, $proxyQuery);
+
         if ('' !== (string) $value) {
-            $this->assertSameQuery(['WHERE alias.field LIKE :field_name_0'], $proxyQuery);
             $this->assertSameQueryParameters(['field_name_0' => sprintf('%%%s%%', $value)], $proxyQuery);
             $this->assertTrue($filter->isActive());
         } else {
-            $this->assertSameQuery([], $proxyQuery);
             $this->assertFalse($filter->isActive());
         }
     }
 
+    public function getValues(): iterable
+    {
+        return [
+            'filter by normal value' => ['asd', false],
+            'not filter by empty string' => ['', false],
+            'filter by empty string' => ['', true],
+            'not filter by null' => [null, false],
+            'filter by null' => [null, true],
+            'not filter by 0' => [0, false],
+            'filter by 0' => [0, true],
+            'not filter by \'0\'' => ['0', false],
+            'filter by \'0\'' => ['0', true],
+        ];
+    }
+
     /**
      * @dataProvider getValues
      */
     public function testContains($value, bool $allowEmpty): void
     {
-        $filter = new StringFilter();
+        $doctrineConfig = $this->createStub(Configuration::class);
```

Comment:
```suggestion
        $doctrineConfig = $this->createMock(Configuration::class);
```

I think this should be mock since the method `with()` is called and is not present in [`InvocationStubber`](https://github.com/sebastianbergmann/phpunit/blob/1b3675760e1a6d3048b53eb092022a67951beaa4/src/Framework/MockObject/Builder/InvocationStubber.php).